From a1038eea3ea5807a322b2b069c674ebb2f30ddbf Mon Sep 17 00:00:00 2001
From: Steven Landow <steven@stlcomputerservices.com>
Date: Mon, 8 Nov 2021 09:33:57 -0800
Subject: enable multicluster headless by default (#35883) (#35950)

* enable multicluster headless by default

Change-Id: If1e6b31e3dafd65fefc81252d162ec534f784d5d

* release note

Change-Id: Ia782ed05a9e12eb6aec054b93526e3685e377b20

* mention dns

Change-Id: I5e00faa8c293fdd193b480673f65aaabfba9431c

* fix

Change-Id: I5324a4ad82b968036fec4713e58024e563fd7923
---
 pilot/pkg/features/pilot.go                            |  2 +-
 releasenotes/notes/35883.yaml                          | 10 ++++------
 releasenotes/notes/35884.yaml                          |  7 +++++++
 .../iop-integration-test-defaults-with-quic.yaml       |  1 -
 tests/integration/iop-integration-test-defaults.yaml   |  1 -
 5 files changed, 12 insertions(+), 9 deletions(-)
 create mode 100644 releasenotes/notes/35884.yaml

diff --git a/pilot/pkg/features/pilot.go b/pilot/pkg/features/pilot.go
index 08735d4a88..9690de8c1a 100644
--- a/pilot/pkg/features/pilot.go
+++ b/pilot/pkg/features/pilot.go
@@ -536,7 +536,7 @@
 	EnableRouteCollapse = env.RegisterBoolVar("PILOT_ENABLE_ROUTE_COLLAPSE_OPTIMIZATION", true,
 		"If true, Pilot will merge virtual hosts with the same routes into a single virtual host, as an optimization.").Get()
 
-	MulticlusterHeadlessEnabled = env.RegisterBoolVar("ENABLE_MULTICLUSTER_HEADLESS", false,
+	MulticlusterHeadlessEnabled = env.RegisterBoolVar("ENABLE_MULTICLUSTER_HEADLESS", true,
 		"If true, the DNS name table for a headless service will resolve to same-network endpoints in any cluster.").Get()
 
 	CertSignerDomain = env.RegisterStringVar("CERT_SIGNER_DOMAIN", "", "The cert signer domain info").Get()
diff --git a/releasenotes/notes/35883.yaml b/releasenotes/notes/35883.yaml
index 521b50381d..5c771c7c23 100644
--- a/releasenotes/notes/35883.yaml
+++ b/releasenotes/notes/35883.yaml
@@ -1,10 +1,8 @@
 apiVersion: release-notes/v2
 kind: feature
-area: istioctl
-issue:
-  - 18487
-
+area: traffic-management
 releaseNotes:
 - |
-  **Updated** `WorkloadEntry` resources will be read across clusters and no longer need to be manually mirrored to other
-  clusters. These endpoints are now automatically discovered and reachable in both multi-cluster and multi-network meshes.
+  **Enabled** discovering cross-cluster, same network, endpoints for headless services when `ISTIO_META_DNS_CAPTURE` and
+  `ISTIO_META_DNS_AUTO_ALLOCATE` are enabled in `ProxyConfig.proxyMetadata`. This can be disabled by setting
+  `ENABLE_MULTICLUSTER_HEADLESS=false` in istiod.
diff --git a/releasenotes/notes/35884.yaml b/releasenotes/notes/35884.yaml
new file mode 100644
index 0000000000..2d0ad21c6a
--- /dev/null
+++ b/releasenotes/notes/35884.yaml
@@ -0,0 +1,7 @@
+apiVersion: release-notes/v2
+kind: feature
+area: traffic-management
+releaseNotes:
+  - |
+    **Updated** `WorkloadEntry` resources will be read across clusters and no longer need to be manually mirrored to other
+    clusters. These endpoints are now automatically discovered and reachable in both multi-cluster and multi-network meshes.
\ No newline at end of file
diff --git a/tests/integration/iop-integration-test-defaults-with-quic.yaml b/tests/integration/iop-integration-test-defaults-with-quic.yaml
index 0b9be45b90..f1914677f0 100644
--- a/tests/integration/iop-integration-test-defaults-with-quic.yaml
+++ b/tests/integration/iop-integration-test-defaults-with-quic.yaml
@@ -60,7 +60,6 @@ spec:
       autoscaleEnabled: false
       env:
         UNSAFE_ENABLE_ADMIN_ENDPOINTS: true
-        ENABLE_MULTICLUSTER_HEADLESS: true
         PILOT_REMOTE_CLUSTER_TIMEOUT: 15s
         PILOT_ENABLE_QUIC_LISTENERS: true
 
diff --git a/tests/integration/iop-integration-test-defaults.yaml b/tests/integration/iop-integration-test-defaults.yaml
index fc24cf686e..a0ff776487 100644
--- a/tests/integration/iop-integration-test-defaults.yaml
+++ b/tests/integration/iop-integration-test-defaults.yaml
@@ -56,7 +56,6 @@ spec:
       autoscaleEnabled: false
       env:
         UNSAFE_ENABLE_ADMIN_ENDPOINTS: true
-        ENABLE_MULTICLUSTER_HEADLESS: true
         PILOT_REMOTE_CLUSTER_TIMEOUT: 15s
 
     gateways:
-- 
2.35.3

